### PR TITLE
EVM ERC20 Template Contract

### PR DESCRIPTION
**Pull request by jewei1997**
## Describe your changes and provide context

Created ERC20 Template contract to allow native sei tokens to be used as ERC20s

## Testing performed to validate your change
foundry unit tests + manual testing.

### CODE REVIEW

File: contracts/src/Counter.sol [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+contract Counter {
```
Comment (codchen) [inline contracts/src/Counter.sol]: do we need this counter contract?

File: contracts/src/NativeSeiTokensERC20.sol [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+contract NativeSeiTokensERC20 is ERC20 {
+
+    address constant BANK_PRECOMPILE_ADDRESS = 0x0000000000000000000000000000000000001001;
+
+    string public denom;
+    IBank public BankPrecompile;
+
+    constructor(string memory denom_) ERC20("", "") {
+        BankPrecompile = IBank(BANK_PRECOMPILE_ADDRESS);
```
Comment (codchen) [inline contracts/src/NativeSeiTokensERC20.sol:16]: just curious: how much space would it take to store `BankPrecompile` as a state? Is it just an address?

File: contracts/src/NativeSeiTokensERC20.sol [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IBank} from "./precompiles/IBank.sol";
+import {console2} from "forge-std/Test.sol";
+
+contract NativeSeiTokensERC20 is ERC20 {
```
Comment (codchen) [inline contracts/src/NativeSeiTokensERC20.sol:8]: `is ERC20` gives us the sending and approval functionalities i suppose?

**Review by codchen — COMMENTED** [review]
overall looks good. We can test it on a local sei before merging:
1. create a tokenfactory denom: `seid tx tokenfactory create-denom test`
2. mint some tokens for an account: `seid tx tokenfactory mint`
3. instantiate an ERC-20 contract for the denom
4. send tokens from the minted account to another account via ERC-20
5. verify the receiver received the tokens

**Review by philipsu522 — APPROVED** [review]

File: contracts/src/NativeSeiTokensERC20.sol (hunk 3 above)
Comment (jewei1997) [reply contracts/src/NativeSeiTokensERC20.sol:8]: yes it is the syntax to do inheritance.

File: contracts/src/NativeSeiTokensERC20.sol (hunk 2 above)
Comment (jewei1997) [reply contracts/src/NativeSeiTokensERC20.sol:16]: yes it should just be an address.

File: contracts/src/Counter.sol (hunk 1 above)
Comment (jewei1997) [reply contracts/src/Counter.sol]: nope, removed.

**Pull request merged**